client/rest-high-level/src/main/java/org/elasticsearch/client/ml/DeleteFilterRequest.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.elasticsearch.client.ml;
+
+import org.elasticsearch.client.Validatable;
+
+import java.util.Objects;
+
+/**
+ * A request to delete a machine learning filter
+ */
+public class DeleteFilterRequest implements Validatable {
+
+    private final String filterId;
+
+    public DeleteFilterRequest(String filterId) {
+        this.filterId = Objects.requireNonNull(filterId, "[filter_id] is required");
```

Comment:
should this still be `[filter_id]` ?

Reply:
@hub-cap yes, that holds with other ML nonNull checks. 

e.g.
https://github.com/elastic/elasticsearch/blob/fc774a377671d9f633513b3f8d954b237b377178/client/rest-high-level/src/main/java/org/elasticsearch/client/ml/job/config/AnalysisConfig.java#L341